consumer.go

```diff
@@ -64,40 +66,73 @@ func (consumer *consumer) Stop() {
 }
 
 type worker struct {
+	sync.Mutex      // Protect channel during reconnect.
+	channel         wabbit.Channel
 	deliveries      <-chan wabbit.Delivery
+	errorChannel    chan<- error
 	shutdownChannel chan struct{}
 	status          int32
 }
 
-func newWorker(deliveries <-chan wabbit.Delivery) worker {
-	return worker{
-		deliveries:      deliveries,
+func newWorker(errorChannel chan<- error) *worker {
+	return &worker{
+		errorChannel:    errorChannel,
 		shutdownChannel: make(chan struct{}),
 	}
 }
 
-func (worker worker) Run(handler ConsumerHandler) {
+func (worker *worker) Run(handler ConsumerHandler) {
 	atomic.StoreInt32(&worker.status, statusRunning)
 
 	for {
 		select {
 		case message := <-worker.deliveries:
-			if message == nil { // Channel has been closed.
-				return
+			if message == nil { // Seems like channel was closed.
+				if worker.changeStatusToStoppedAtomic() {
```

Comment:
boolean should answer yes/no question so 
worker.**hasChangedStatusToStoppedAtomic**()